### PR TITLE
Fix runner loader

### PR DESCRIPTION
**Pull request by m4dcoder**
The frequent reloading of runner modules at runtime leads to random errors complaining the self instance of super(RunnerType, self).pre_run(...) is not instance type of the runner type. This PR address this issue by loading the runner modules only once and storing a copy of the module for reuse.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -49,3 +49,4 @@ nosetests.xml
 
 # st2 specific
 contrib/runners/*/dist_utils.py
+st2common/tests/fixturesc
```
Comment (lakshmi-kannan) [inline .gitignore]: Typo? Should be /fixtures?

Comment (m4dcoder) [inline .gitignore]: It is not typo.  This is actually generated for whatever reason...

**Comment by Kami** [comment]
That's really weird - I could never reproduce it, but it could be due to local Python module cache or similar.

One thing which I just reminded we could and probably should do is call `.reload()` on the module after we load it.

/cc @bigmstone

File: st2common/st2common/util/loader.py [hunk 2]
```diff
@@ -176,9 +178,10 @@ def register_runner(module_name):
 
     LOG.debug('Loading runner from: %s', module_path)
 
-    module = imp.load_source(module_name, module_path)
+    if module_name not in RUNNER_MODULES:
```
Comment (Kami) [inline st2common/st2common/util/loader.py]: Yeah, I'm personally fine with caching an existing module. It's also worth noting that this requires process restart (action runner) in case module changes and user wants those changes to take an effect, but imo that's fine and just something we need to document.

One thing we could probably to avoid the need of restart (well in most cases) or avoid the need to cache is to use `.reload()` (or delete module from internal module cache), but that's not Erlang and it's not really that robust and good idea to use it in most cases so process restart is more robust.

Anyway, just something to document in pluggable runner section of documentation :P

/cc @bigmstone

File: st2common/st2common/util/loader.py [hunk 3]
```diff
@@ -189,9 +192,10 @@ def register_query_module(module_name):
 
     LOG.debug('Loading query module from: %s', module_path)
 
-    module = imp.load_source(module_name, module_path)
+    if module_name not in QUERIER_MODULES:
+        QUERIER_MODULES[module_name] = imp.load_source(module_name, module_path)
```
Comment (Kami) [inline st2common/st2common/util/loader.py]: Let's please add an `else` and LOG.info / debug statement which says that module x for runner y is already cached - this way user won't be that surprised if they made a change without restarting action runner and wondering why the change is not taking effect.

**Comment by Kami** [comment]
I added some comments - nice catch and good work overall :+1: 

There are just some things we need to document and all be aware of since it does change the behavior a bit (restart vs no restart, etc.).

**Comment by jbrownsc** [comment]
I'm seeing an issue that seems very similar to this one. This stackstorm env hasn't had any executions in a while.  The first run of of a core.local action raised an exception around LocalShellRunner, and the first python action (st2.kv.set) raised an exception around PythonRunner.    I suspect something is getting garbage collected or some lower level construct that I don't understand is cleaning something up.  If I re-run the execution, it works just fine.

```super(type, obj): obj must be an instance or subtype of type
  File "/opt/stackstorm/st2/local/lib/python2.7/site-packages/st2actions/container/base.py", line 110, in _do_run
    runner.pre_run()
  File "/opt/stackstorm/runners/local_runner/local_runner.py", line 84, in pre_run
    super(LocalShellRunner, self).pre_run()

super(type, obj): obj must be an instance or subtype of type
File "/opt/stackstorm/st2/local/lib/python2.7/site-packages/st2actions/container/base.py", line 110, in _do_run
runner.pre_run()
File "/opt/stackstorm/runners/python_runner/python_runner.py", line 86, in pre_run
super(PythonRunner, self).pre_run()```
